app/services/create_new_device_alert.rb

```diff
@@ -0,0 +1,45 @@
+class CreateNewDeviceAlert < ApplicationJob
+  queue_as :long_running
+
+  def perform(now)
+    emails_sent = 0
+    User.where(
+      sql_query_for_users_with_new_device,
+      tvalue: now - IdentityConfig.store.new_device_alert_delay_in_minutes.minutes,
+    ).each do |user|
+      emails_sent += 1 if clear_new_device_and_send_email(user)
+    end
+
+    emails_sent
+  end
+
+  private
+
+  def sql_query_for_users_with_new_device
+    <<~SQL
+      sign_in_new_device_at < :tvalue AND
```

Comment:
Can we include a test scenario to verify that a user won't be touched if `sign_in_new_device_at` is `nil` ?

It's not immediately obvious to me the behavior of this query if `sign_in_new_device_at` is `nil`.

Reply:
I have added a test for nil.